docs/introduction/how-to-work-with-money.md

```diff
@@ -0,0 +1,388 @@
+# How to Work with Money
+
+Money is a very important concept for every e-commerce project.
+In Shopsys Framework, all monetary values (*prices, account balances, discount amounts, price limits etc.*) are represented by an instance of [the `Money` class](/packages/framework/src/Component/Money/Money.php).
+
+This approach has several advantages:
+- it avoids problems with floating point number calculations and comparisons (see [official PHP documentation](http://php.net/manual/en/language.types.float.php) for details)
+- allows easy-to-use interfaces with consistent type-hinting so you can be sure what type of value you should be using
+- prevents accidental conversion to unexpected types (which may be problematic eg. when using the `===` operator)
+- makes the application design clearer and future changes easier
+
+**Table of Contents:**
+- [Money class](#money-class)
+- [Money in Forms](#money-in-forms)
+- [Money in Twig Templates](#money-in-twig-templates)
+- [Money in Javascript](#money-in-javascript)
+- [Money in Doctrine](#money-in-doctrine)
+- [Unit and Functional Tests](#unit-and-functional-tests)
+- [Price class](#price-class)
+
```

Comment:
After a short break I think the `scale` concept should be explained at the top of the article. It it a tricky concept and I would appreciate conceptual explanation with examples. Like if you divide 1/3, you have to specify the scale and if you pick 2, the result will be `1.33`. You have to specify scale for money creating and if you create money from 1.494 it has different results for scale 1 - `1.5` and 2 - `1.49`. Also explain here that scale rounds the number and how

Reply:
The scale might deserve its own section, but I wouldn't want to explain the WHATs and WHYs any further. I think that the introductory section covers it sufficiently in the scope of this PR:

> Money is a very important concept for every e-commerce project. In Shopsys Framework, all monetary values (prices, account balances, discount amounts, price limits etc.) are represented by an instance of the Money class.
>
> This approach has several advantages:
>
> - it avoids problems with floating point number calculations and comparisons (see official PHP documentation for details)
> - allows easy-to-use interfaces with consistent type-hinting so you can be sure what type of value you should be using
> - prevents accidental conversion to unexpected types (which may be problematic eg. when using the === operator)
> - makes the application design clearer and future changes easier

That currency is not included in the value object is explained in "Money class" section:

> It uses a decimal representation of the money amount and it does not contain any reference to the used currency.

I had a hard time explaining the resons in a clear way because in the end, we'd actually like to include it but we could not implement it in the possible timeframe (https://github.com/shopsys/shopsys/issues/745#issuecomment-460635377). We've discussed this with @vitek-rostislav in https://github.com/shopsys/shopsys/pull/870#discussion_r261502306 as well. Do you have any ideas how to explain it without breaking the flow or getting into nitty gritty details?

Reply:
**Suggestion:**

## Concept

The money concept in SSFW represents and encapsulates money values with decimal part, like `100`, `0.5`, `10.99`, `0.0005`. Money are representation without currency (if you need to pass also information about currency through the system, you need a separated concept).

### Scale
Scale is a precision after decimal part and it is a bit tricky concept.

Imagine you want to represent `1/3`, in float it is actually repsesented as `0.3333333432674407958984375` (because of precision (link)). When you want to work with the `1/3` in terms of money, you have to specify "amount of places after decimal point". So you can create a money from `1/3` in scale of `2` -> `0.33`, or in scale of `8` -> `0.33333334`

 Scale have to be specified during rounding (link), creating from floats (link) or dividing (link).

Reply:
I've added the section in aa1e1900c